src/datastore/postgres-store.ts

```diff
@@ -4543,8 +4543,7 @@ export class PgDataStore
         `
       SELECT tx_id, canonical, contract_id, block_height, source_code, abi
       FROM smart_contracts
-      WHERE contract_id = ANY($1)
-      ORDER BY abi != 'null' DESC, canonical DESC, microblock_canonical DESC, block_height DESC
+      WHERE contract_id = ANY($1) AND abi != '' AND abi IS NOT NULL AND canonical = true AND microblock_canonical = true
```

Comment:
This change has the side effect of no longer returning smart contract data that is orphaned or micro-orphaned. The previous query used `ORDER BY` to _prefer_ returning the canonical result, if available. However, the recent change to query for many smart contracts was missing an `ORDER BY contract_id` and a `DISTINCT ON(contract_id)`.